src/Corax/IndexFieldsMapping.cs

```diff
@@ -30,11 +30,37 @@ public IndexFieldsMapping(ByteStringContext context)
         _fieldsList = new List<IndexFieldBinding>();
     }
 
+    
+    private const int I64Postfix = 875972909; //"-I64"
+    private const int DblPostfix = 1818379309; //"-Dbl"
+
+    public ByteStringContext<ByteStringMemoryCache>.InternalScope GetFieldNameForLongs(Slice fieldName, out Slice fieldNameForLongs)
+    {
+        return GetFieldNameWithPostfix(fieldName, I64Postfix, out fieldNameForLongs);
+    }
+    
+    public ByteStringContext<ByteStringMemoryCache>.InternalScope GetFieldNameForDoubles(Slice fieldName, out Slice fieldNameForDoubles)
+    {
+        return GetFieldNameWithPostfix(fieldName, DblPostfix, out fieldNameForDoubles);
+    }
+
+    private unsafe ByteStringContext<ByteStringMemoryCache>.InternalScope GetFieldNameWithPostfix(Slice fieldName, int postfix, out Slice fieldWithPostfix)
+    {
+        var scope = _context.Allocate(fieldName.Size + sizeof(int), out ByteString output);
+        fieldName.Content.CopyTo(output.Ptr);
+        *(int*)(output.Ptr + fieldName.Size) = postfix;
+        fieldWithPostfix = new Slice(SliceOptions.Key, output);
+        return scope;
+    }
+
     public IndexFieldsMapping AddBinding(int fieldId, Slice fieldName, Analyzer analyzer = null, bool hasSuggestion = false, FieldIndexingMode fieldIndexingMode = FieldIndexingMode.Normal, bool hasSpatial = false)
     {
         if (!_fieldsById.TryGetValue(fieldId, out var storedAnalyzer))
         {
-            var binding = new IndexFieldBinding(fieldId, fieldName, analyzer, hasSuggestion, fieldIndexingMode, hasSpatial);
+            GetFieldNameForDoubles(fieldName, out var fieldNameDouble);
+            GetFieldNameForLongs(fieldName, out var fieldNameLong);
```

Comment:
@maciejaszyk Here we _have_ to check that the lifetime of the `IndexFeidlsMapping` is as long as the `context`.
I'm fairly certain that it is, but I want another pair of eyes on that.